src/views/Leads/Grid/index.js

```diff
@@ -0,0 +1,173 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import Masonry from '#rc/Masonry';
+import Button from '#rsca/Button';
+import LoadingAnimation from '#rscv/LoadingAnimation';
+import { isArrayEqual } from '#rsu/common';
+import Modal from '#rscv/Modal';
+import LeadPreview from '#components/leftpanel/LeadPreview';
+import { iconNames } from '#constants';
+import {
+    leadsForProjectGridViewSelector,
+} from '#redux';
+import LeadItem from '../GridItem';
+import styles from './styles.scss';
+
+const propTypes = {
+    loading: PropTypes.bool,
+    onEndReached: PropTypes.func.isRequired,
+    leads: PropTypes.arrayOf(PropTypes.object),
+    view: PropTypes.string.isRequired,
+    activeProject: PropTypes.number.isRequired,
+    onSearchSimilarLead: PropTypes.func.isRequired,
+    onRemoveLead: PropTypes.func.isRequired,
+    onMarkProcessed: PropTypes.func.isRequired,
+    onMarkPending: PropTypes.func.isRequired,
+    setLeadPageActivePage: PropTypes.func.isRequired,
+    emptyComponent: PropTypes.func.isRequired,
+};
+
+const defaultProps = {
+    loading: false,
+    leads: [],
+};
+
+const LEFT_KEY = 37;
+const RIGHT_KEY = 39;
+
+const mapStateToProps = state => ({
+    leads: leadsForProjectGridViewSelector(state),
+});
+
+@connect(mapStateToProps)
+export default class LeadGrid extends React.Component {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    // simple comparision using id and their position to
+    // check if we should recompute the layout
+    // we can add thumbnail url check too if the leads data
+    // is updated in realtime using websockets
+    static isItemsChanged = (a, b) => {
+        const aKeys = a.map(r => r.id);
+        const bKeys = b.map(r => r.id);
+
+        return !isArrayEqual(aKeys, bKeys);
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showPreview: false,
+            activeLeadIndex: 0,
+        };
+
+        this.columnWidth = 300;
+        this.columnGutter = 40;
+
+        this.itemState = {
+            width: this.columnWidth,
+            minHeight: 295,
+        };
+
+        this.masonryRef = {};
+    }
+
+    componentDidMount() {
+        // Rest grid when loading first time
+        this.props.setLeadPageActivePage({ activePage: 1 });
+    }
+
+    shouldComponentUpdate() {
```

Comment:
Can you explain the use of shouldComponentUpdate here?

Reply:
This was to prevent the recomputation in Grid when the active view is Table but is not needed now after separation of states of table and grid.